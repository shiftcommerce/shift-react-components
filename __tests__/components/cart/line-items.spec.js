// Libraries
import React from 'react'

import Image from '../../../src/objects/image'
import LineItems from '../../../src/components/cart/line-items'

import cartFixture from '../../fixtures/cart'

test('renders line items correctly, on initial load', () => {
  // Act
  const wrapper = shallow(
    <LineItems
      lineItems={cartFixture.line_items}
      lineItemsCount={cartFixture.line_items_count}
    />
  )

  // Assert
  const lineItem = cartFixture.line_items[0]
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(lineItem.item.product.title)
  expect(wrapper).toContainReact(<Image className='c-line-items__image' src={lineItem.item.picture_url} alt={lineItem.item.title} key={lineItem.item.product.slug} aria-label={lineItem.item.title} />)
  expect(wrapper.find('select').props().value).toBe(lineItem.unit_quantity)
  expect(wrapper).toIncludeText(lineItem.sku)
  expect(wrapper).toIncludeText((lineItem.item.price * lineItem.unit_quantity) - (lineItem.total_discount))
})

test('renders line items sorted by their id', () => {
  // Arrange
  const lineItems = [
    {
      id: '311',
      item: {
        picture_url: 'https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg',
        product: {
          canonical_path: '/jugs/seed_product_40',
          slug: 'jugs/seed_product_40',
          title: 'First product'
        },
        title: 'First variant'
      },
      item_id: 321,
      item_type: 'Variant',
      line_item_discounts: [
        {
          id: '2a720d8d-ae39-462e-b2e2-4c49ca5c8846',
          line_item_number: 1,
          promotion_id: 21,
          total: 1.5
        }
      ],
      sku: '3008557600817',
      stock_available_level: 85,
      sub_total: 6.77,
      total: 5.27,
      total_discount: 1.5,
      unit_quantity: 1
    },
    {
      id: '310',
      item: {
        picture_url: 'https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg',
        product: {
          canonical_path: '/jugs/seed_product_39',
          slug: 'jugs/seed_product_39',
          title: 'Second product'
        },
        title: 'Second variant'
      },
      item_id: 321,
      item_type: 'Variant',
      line_item_discounts: [
        {
          id: '2a720d8d-ae39-462e-b2e2-4c49ca5c8846',
          line_item_number: 1,
          promotion_id: 21,
          total: 1.5
        }
      ],
      sku: '3008557600817',
      stock_available_level: 85,
      sub_total: 6.77,
      total: 5.27,
      total_discount: 1.5,
      unit_quantity: 1
    }
  ]

  // Act
  const wrapper = shallow(
    <LineItems
      lineItems={lineItems}
      lineItemsCount={2}
    />
  )

  // Assert
  const firstItem = wrapper.find('.c-line-items__sections').first()
  expect(firstItem).toIncludeText('Second product - Second variant')
})

test('trigger updateQuantity function, on change of line item quantity', () => {
  const updateQuantity = jest.fn()

  // act
  const wrapper = shallow(
    <LineItems
      lineItems={cartFixture.line_items}
      lineItemsCount={cartFixture.line_items_count}
      updateQuantity={updateQuantity}
    />
  )

  // assert
  wrapper.find('select').simulate('change', { target: { value: 3 } })
  expect(wrapper).toMatchSnapshot()
  expect(updateQuantity).toHaveBeenCalled()
})
