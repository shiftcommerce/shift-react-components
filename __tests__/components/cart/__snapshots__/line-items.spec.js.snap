// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders line items correctly, on initial load 1`] = `
<LineItems
  lineItems={
    Array [
      Object {
        "cart_id": 151,
        "container_id": 151,
        "gift": false,
        "id": "311",
        "item": Object {
          "available_to_browse": true,
          "available_to_order": true,
          "created_at": "2018-09-24T15:43:00Z",
          "current_price": 6.77,
          "deleted_at": null,
          "description": "Product 40 - variant 1 Grown at an altitude of 4, 500 feet, the tea bush 'wakes' after the winter in April and produces delicate white buds, which often have a white down on them.They are picked by hand and sun- dried so that fermentation cannot take place. Very little white tea is produced, as this process requires particular care, increasing its rarity and making it highly sought after by tea buyers all over the world. It has a delicate flavour, so is best drunk without milk.",
          "ewis_eligible": false,
          "has_assets": true,
          "id": "321",
          "meta_attributes": Object {},
          "ordered_total": 10,
          "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
          "price": 6.77,
          "price_includes_taxes": true,
          "product": Object {
            "attribute_merchandising": Object {},
            "available_to_browse": true,
            "average_rating": 0,
            "canonical_path": "/jugs/seed_product_40",
            "category_ids_cache": Array [
              57,
            ],
            "content_type": "html",
            "created_at": "2018-09-24T15:43:00Z",
            "description": "Grown at an altitude of 4,500 feet, the tea bush 'wakes' after the winter in April and produces delicate white buds, which often have a white down on them. They are picked by hand and sun-dried so that fermentation cannot take place. Very little white tea is produced, as this process requires particular care, increasing its rarity and making it highly sought after by tea buyers all over the world. It has a delicate flavour, so is best drunk without milk.",
            "ewis_status": "not_ewis_eligible",
            "has_assets": true,
            "id": "164",
            "max_current_price": 90.31,
            "max_price": 90.31,
            "merchandised": true,
            "merged_attribute_merchandising": Object {},
            "meta_attributes": Object {},
            "min_current_price": 6.77,
            "min_price": 6.77,
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
            "published": true,
            "rating": 0,
            "reference": "seed_product_40",
            "sellable": true,
            "seo_meta_description": "Product 40 - Qui sed maiores occaecati.",
            "seo_meta_keywords": "minus, ut, ut, adipisci, consequatur",
            "seo_meta_title": "SCS-04-CJ coffee jug 600ml",
            "slug": "jugs/seed_product_40",
            "title": "SCS-04-CJ coffee jug 600ml",
            "updated_at": "2018-11-21T10:16:59Z",
            "web_product_description": "Product 40 - Totam aut quo facere.",
            "web_product_name": "SCS-04-CJ coffee jug 600ml",
          },
          "product_id": 164,
          "reference": "est_36_1",
          "sku": "3008557600817",
          "stock_allocated_level": 15,
          "stock_available_level": 85,
          "stock_level": 100,
          "tax_code": "vat",
          "title": "Variant 1",
          "updated_at": "2018-11-20T13:18:46Z",
        },
        "item_id": 321,
        "item_type": "Variant",
        "line_item_discounts": Array [
          Object {
            "id": "2a720d8d-ae39-462e-b2e2-4c49ca5c8846",
            "line_item_number": 1,
            "promotion_id": 21,
            "total": 1.5,
          },
        ],
        "sku": "3008557600817",
        "stock_available_level": 85,
        "sub_total": 6.77,
        "tax": 0.88,
        "tax_rate": "0.2",
        "title": "Variant 1",
        "total": 5.27,
        "total_discount": 1.5,
        "unit_price": 6.77,
        "unit_quantity": 1,
      },
    ]
  }
  lineItemsCount={2}
  updateQuantity={[MockFunction]}
>
  <div
    className="c-line-items"
  >
    <div
      className="c-line-items__sections"
      key="3008557600817"
    >
      <div
        className="c-line-items__images"
      >
        <Link
          href="/slug?slug=/jugs/seed_product_40"
        >
          <a
            href="/slug?slug=/jugs/seed_product_40"
          >
            <Image
              alt="Variant 1"
              aria-label="Variant 1"
              className="c-line-items__image"
              key="jugs/seed_product_40"
              src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg"
            >
              <div
                className="c-line-items__image"
              >
                <picture>
                  <img
                    alt="Variant 1"
                    aria-label="Variant 1"
                    className="o-image"
                    src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg"
                    srcSet="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg 769w"
                  />
                </picture>
              </div>
            </Image>
          </a>
        </Link>
      </div>
      <div
        className="c-line-items__information"
      >
        <div
          className="c-line-items__title"
        >
          <div
            className="c-line-items__details"
          >
            <h4
              className="c-line-items__details-title u-bold"
            >
              SCS-04-CJ coffee jug 600ml - Variant 1
            </h4>
            <div
              className="c-line-items__details-sku"
            >
              <span>
                3008557600817
              </span>
            </div>
          </div>
          <div
            className="c-line-items__remove"
          >
            <div
              className="c-line-items__delete"
            >
              <a
                className="c-line-items__delete-button"
                data-id="311"
              >
                Delete
              </a>
            </div>
          </div>
        </div>
        <div
          className="c-line-items__quantity"
        >
          <div
            className="c-line-items__quantity-header"
          >
            <span>
              Quantity
            </span>
          </div>
          <_class
            className="c-line-items__quantity-select"
            data-id="311"
            label="Quantity"
            onChange={[MockFunction]}
            options={
              Array [
                Object {
                  "title": 1,
                  "value": 1,
                },
                Object {
                  "title": 2,
                  "value": 2,
                },
                Object {
                  "title": 3,
                  "value": 3,
                },
                Object {
                  "title": 4,
                  "value": 4,
                },
                Object {
                  "title": 5,
                  "value": 5,
                },
                Object {
                  "title": 6,
                  "value": 6,
                },
                Object {
                  "title": 7,
                  "value": 7,
                },
                Object {
                  "title": 8,
                  "value": 8,
                },
                Object {
                  "title": 9,
                  "value": 9,
                },
                Object {
                  "title": 10,
                  "value": 10,
                },
              ]
            }
            skipLabel={true}
            skipPrompt={true}
            value={1}
          >
            <DropdownSelect
              className="c-line-items__quantity-select"
              data-id="311"
              label="Quantity"
              onChange={[MockFunction]}
              options={
                Array [
                  Object {
                    "title": 1,
                    "value": 1,
                  },
                  Object {
                    "title": 2,
                    "value": 2,
                  },
                  Object {
                    "title": 3,
                    "value": 3,
                  },
                  Object {
                    "title": 4,
                    "value": 4,
                  },
                  Object {
                    "title": 5,
                    "value": 5,
                  },
                  Object {
                    "title": 6,
                    "value": 6,
                  },
                  Object {
                    "title": 7,
                    "value": 7,
                  },
                  Object {
                    "title": 8,
                    "value": 8,
                  },
                  Object {
                    "title": 9,
                    "value": 9,
                  },
                  Object {
                    "title": 10,
                    "value": 10,
                  },
                ]
              }
              renderValidationMessage={[Function]}
              skipLabel={true}
              skipPrompt={true}
              value={1}
            >
              <div
                className="o-form__input-group"
              >
                <select
                  className="o-form__input-block o-form__input-field c-line-items__quantity-select"
                  data-id="311"
                  label="Quantity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  role="listbox"
                  value={1}
                >
                  <option
                    aria-posinset={1}
                    aria-setsize={10}
                    key="0"
                    role="option"
                    value={1}
                  >
                    1
                  </option>
                  <option
                    aria-posinset={2}
                    aria-setsize={10}
                    key="1"
                    role="option"
                    value={2}
                  >
                    2
                  </option>
                  <option
                    aria-posinset={3}
                    aria-setsize={10}
                    key="2"
                    role="option"
                    value={3}
                  >
                    3
                  </option>
                  <option
                    aria-posinset={4}
                    aria-setsize={10}
                    key="3"
                    role="option"
                    value={4}
                  >
                    4
                  </option>
                  <option
                    aria-posinset={5}
                    aria-setsize={10}
                    key="4"
                    role="option"
                    value={5}
                  >
                    5
                  </option>
                  <option
                    aria-posinset={6}
                    aria-setsize={10}
                    key="5"
                    role="option"
                    value={6}
                  >
                    6
                  </option>
                  <option
                    aria-posinset={7}
                    aria-setsize={10}
                    key="6"
                    role="option"
                    value={7}
                  >
                    7
                  </option>
                  <option
                    aria-posinset={8}
                    aria-setsize={10}
                    key="7"
                    role="option"
                    value={8}
                  >
                    8
                  </option>
                  <option
                    aria-posinset={9}
                    aria-setsize={10}
                    key="8"
                    role="option"
                    value={9}
                  >
                    9
                  </option>
                  <option
                    aria-posinset={10}
                    aria-setsize={10}
                    key="9"
                    role="option"
                    value={10}
                  >
                    10
                  </option>
                </select>
              </div>
            </DropdownSelect>
          </_class>
        </div>
        <div
          className="c-line-items__amounts"
        >
          <a
            className="c-line-items__amount"
          >
            £
            6.77
          </a>
          <a
            className="c-line-items__amount c-line-items__amount--discount"
          >
            - £
            1.50
          </a>
          <a
            className="c-line-items__amount c-line-items__amount--total"
          >
            £
            5.27
          </a>
        </div>
      </div>
    </div>
  </div>
</LineItems>
`;

exports[`trigger updateQuantity function, on change of line item quantity 1`] = `
<LineItems
  lineItems={
    Array [
      Object {
        "cart_id": 151,
        "container_id": 151,
        "gift": false,
        "id": "311",
        "item": Object {
          "available_to_browse": true,
          "available_to_order": true,
          "created_at": "2018-09-24T15:43:00Z",
          "current_price": 6.77,
          "deleted_at": null,
          "description": "Product 40 - variant 1 Grown at an altitude of 4, 500 feet, the tea bush 'wakes' after the winter in April and produces delicate white buds, which often have a white down on them.They are picked by hand and sun- dried so that fermentation cannot take place. Very little white tea is produced, as this process requires particular care, increasing its rarity and making it highly sought after by tea buyers all over the world. It has a delicate flavour, so is best drunk without milk.",
          "ewis_eligible": false,
          "has_assets": true,
          "id": "321",
          "meta_attributes": Object {},
          "ordered_total": 10,
          "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
          "price": 6.77,
          "price_includes_taxes": true,
          "product": Object {
            "attribute_merchandising": Object {},
            "available_to_browse": true,
            "average_rating": 0,
            "canonical_path": "/jugs/seed_product_40",
            "category_ids_cache": Array [
              57,
            ],
            "content_type": "html",
            "created_at": "2018-09-24T15:43:00Z",
            "description": "Grown at an altitude of 4,500 feet, the tea bush 'wakes' after the winter in April and produces delicate white buds, which often have a white down on them. They are picked by hand and sun-dried so that fermentation cannot take place. Very little white tea is produced, as this process requires particular care, increasing its rarity and making it highly sought after by tea buyers all over the world. It has a delicate flavour, so is best drunk without milk.",
            "ewis_status": "not_ewis_eligible",
            "has_assets": true,
            "id": "164",
            "max_current_price": 90.31,
            "max_price": 90.31,
            "merchandised": true,
            "merged_attribute_merchandising": Object {},
            "meta_attributes": Object {},
            "min_current_price": 6.77,
            "min_price": 6.77,
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
            "published": true,
            "rating": 0,
            "reference": "seed_product_40",
            "sellable": true,
            "seo_meta_description": "Product 40 - Qui sed maiores occaecati.",
            "seo_meta_keywords": "minus, ut, ut, adipisci, consequatur",
            "seo_meta_title": "SCS-04-CJ coffee jug 600ml",
            "slug": "jugs/seed_product_40",
            "title": "SCS-04-CJ coffee jug 600ml",
            "updated_at": "2018-11-21T10:16:59Z",
            "web_product_description": "Product 40 - Totam aut quo facere.",
            "web_product_name": "SCS-04-CJ coffee jug 600ml",
          },
          "product_id": 164,
          "reference": "est_36_1",
          "sku": "3008557600817",
          "stock_allocated_level": 15,
          "stock_available_level": 85,
          "stock_level": 100,
          "tax_code": "vat",
          "title": "Variant 1",
          "updated_at": "2018-11-20T13:18:46Z",
        },
        "item_id": 321,
        "item_type": "Variant",
        "line_item_discounts": Array [
          Object {
            "id": "2a720d8d-ae39-462e-b2e2-4c49ca5c8846",
            "line_item_number": 1,
            "promotion_id": 21,
            "total": 1.5,
          },
        ],
        "sku": "3008557600817",
        "stock_available_level": 85,
        "sub_total": 6.77,
        "tax": 0.88,
        "tax_rate": "0.2",
        "title": "Variant 1",
        "total": 5.27,
        "total_discount": 1.5,
        "unit_price": 6.77,
        "unit_quantity": 1,
      },
    ]
  }
  lineItemsCount={2}
  updateQuantity={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "target": Object {
              "value": Object {},
            },
          },
          undefined,
          undefined,
          Object {},
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="c-line-items"
  >
    <div
      className="c-line-items__sections"
      key="3008557600817"
    >
      <div
        className="c-line-items__images"
      >
        <Link
          href="/slug?slug=/jugs/seed_product_40"
        >
          <a
            href="/slug?slug=/jugs/seed_product_40"
          >
            <Image
              alt="Variant 1"
              aria-label="Variant 1"
              className="c-line-items__image"
              key="jugs/seed_product_40"
              src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg"
            >
              <div
                className="c-line-items__image"
              >
                <picture>
                  <img
                    alt="Variant 1"
                    aria-label="Variant 1"
                    className="o-image"
                    src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg"
                    srcSet="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg 769w"
                  />
                </picture>
              </div>
            </Image>
          </a>
        </Link>
      </div>
      <div
        className="c-line-items__information"
      >
        <div
          className="c-line-items__title"
        >
          <div
            className="c-line-items__details"
          >
            <h4
              className="c-line-items__details-title u-bold"
            >
              SCS-04-CJ coffee jug 600ml - Variant 1
            </h4>
            <div
              className="c-line-items__details-sku"
            >
              <span>
                3008557600817
              </span>
            </div>
          </div>
          <div
            className="c-line-items__remove"
          >
            <div
              className="c-line-items__delete"
            >
              <a
                className="c-line-items__delete-button"
                data-id="311"
              >
                Delete
              </a>
            </div>
          </div>
        </div>
        <div
          className="c-line-items__quantity"
        >
          <div
            className="c-line-items__quantity-header"
          >
            <span>
              Quantity
            </span>
          </div>
          <_class
            className="c-line-items__quantity-select"
            data-id="311"
            label="Quantity"
            onChange={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "target": Object {
                        "value": Object {},
                      },
                    },
                    undefined,
                    undefined,
                    Object {},
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
            options={
              Array [
                Object {
                  "title": 1,
                  "value": 1,
                },
                Object {
                  "title": 2,
                  "value": 2,
                },
                Object {
                  "title": 3,
                  "value": 3,
                },
                Object {
                  "title": 4,
                  "value": 4,
                },
                Object {
                  "title": 5,
                  "value": 5,
                },
                Object {
                  "title": 6,
                  "value": 6,
                },
                Object {
                  "title": 7,
                  "value": 7,
                },
                Object {
                  "title": 8,
                  "value": 8,
                },
                Object {
                  "title": 9,
                  "value": 9,
                },
                Object {
                  "title": 10,
                  "value": 10,
                },
              ]
            }
            skipLabel={true}
            skipPrompt={true}
            value={1}
          >
            <DropdownSelect
              className="c-line-items__quantity-select"
              data-id="311"
              label="Quantity"
              onChange={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "target": Object {
                          "value": Object {},
                        },
                      },
                      undefined,
                      undefined,
                      Object {},
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                }
              }
              options={
                Array [
                  Object {
                    "title": 1,
                    "value": 1,
                  },
                  Object {
                    "title": 2,
                    "value": 2,
                  },
                  Object {
                    "title": 3,
                    "value": 3,
                  },
                  Object {
                    "title": 4,
                    "value": 4,
                  },
                  Object {
                    "title": 5,
                    "value": 5,
                  },
                  Object {
                    "title": 6,
                    "value": 6,
                  },
                  Object {
                    "title": 7,
                    "value": 7,
                  },
                  Object {
                    "title": 8,
                    "value": 8,
                  },
                  Object {
                    "title": 9,
                    "value": 9,
                  },
                  Object {
                    "title": 10,
                    "value": 10,
                  },
                ]
              }
              renderValidationMessage={[Function]}
              skipLabel={true}
              skipPrompt={true}
              value={1}
            >
              <div
                className="o-form__input-group"
              >
                <select
                  className="o-form__input-block o-form__input-field c-line-items__quantity-select"
                  data-id="311"
                  label="Quantity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  role="listbox"
                  value={1}
                >
                  <option
                    aria-posinset={1}
                    aria-setsize={10}
                    key="0"
                    role="option"
                    value={1}
                  >
                    1
                  </option>
                  <option
                    aria-posinset={2}
                    aria-setsize={10}
                    key="1"
                    role="option"
                    value={2}
                  >
                    2
                  </option>
                  <option
                    aria-posinset={3}
                    aria-setsize={10}
                    key="2"
                    role="option"
                    value={3}
                  >
                    3
                  </option>
                  <option
                    aria-posinset={4}
                    aria-setsize={10}
                    key="3"
                    role="option"
                    value={4}
                  >
                    4
                  </option>
                  <option
                    aria-posinset={5}
                    aria-setsize={10}
                    key="4"
                    role="option"
                    value={5}
                  >
                    5
                  </option>
                  <option
                    aria-posinset={6}
                    aria-setsize={10}
                    key="5"
                    role="option"
                    value={6}
                  >
                    6
                  </option>
                  <option
                    aria-posinset={7}
                    aria-setsize={10}
                    key="6"
                    role="option"
                    value={7}
                  >
                    7
                  </option>
                  <option
                    aria-posinset={8}
                    aria-setsize={10}
                    key="7"
                    role="option"
                    value={8}
                  >
                    8
                  </option>
                  <option
                    aria-posinset={9}
                    aria-setsize={10}
                    key="8"
                    role="option"
                    value={9}
                  >
                    9
                  </option>
                  <option
                    aria-posinset={10}
                    aria-setsize={10}
                    key="9"
                    role="option"
                    value={10}
                  >
                    10
                  </option>
                </select>
              </div>
            </DropdownSelect>
          </_class>
        </div>
        <div
          className="c-line-items__amounts"
        >
          <a
            className="c-line-items__amount"
          >
            £
            6.77
          </a>
          <a
            className="c-line-items__amount c-line-items__amount--discount"
          >
            - £
            1.50
          </a>
          <a
            className="c-line-items__amount c-line-items__amount--total"
          >
            £
            5.27
          </a>
        </div>
      </div>
    </div>
  </div>
</LineItems>
`;
